qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,335 @@ class Command(BaseModel):
     full description of the model, please check ``CommandSchema``.
 
     Attributes:
-        name (str): Pulse command name.
+        name: Pulse command name.
     """
-    def __init__(self, name, **kwargs):
+    def __init__(self, name: str, **kwargs):
         self.name = name
 
         super().__init__(**kwargs)
 
 
 @bind_schema(PulseDefaultsSchema)
 class PulseDefaults(BaseModel):
-    """Model for PulseDefaults.
-
-    Please note that this class only describes the required fields. For the
-    full description of the model, please check ``PulseDefaultsSchema``.
-
-    Attributes:
-        qubit_freq_est (list[number]): Estimated qubit frequencies in GHz.
-        meas_freq_est (list[number]): Estimated measurement cavity frequencies
-            in GHz.
-        buffer (int): Default buffer time (in units of dt) between pulses.
-        pulse_library (list[PulseLibraryItem]): Backend pulse library.
-        cmd_def (list[Command]): Backend command definition.
+    """Description of default settings for Pulse systems. These are operations or settings that
+    may be good starting points for the Pulse user. The user may modify these defaults through
+    the provided methods to build a reference to custom operations, which may in turn be used
+    for building Schedules or converting circuits to Schedules.
     """
 
-    def __init__(self, qubit_freq_est, meas_freq_est, buffer,
-                 pulse_library, cmd_def, **kwargs):
-        self.qubit_freq_est = qubit_freq_est
-        self.meas_freq_est = meas_freq_est
+    def __init__(self,
+                 qubit_freq_est: List[float],
+                 meas_freq_est: List[float],
+                 buffer: int,
+                 pulse_library: List[PulseLibraryItem],
+                 cmd_def: List[Command],
+                 **kwargs: Dict[str, Any]):
+        """
+        Validate and reformat transport layer inputs to initialize.
+
+        Args:
+            qubit_freq_est: Estimated qubit frequencies in GHz.
+            meas_freq_est: Estimated measurement cavity frequencies in GHz.
+            buffer: Default buffer time (in units of dt) between pulses.
+            pulse_library: Pulse name and sample definitions.
+            cmd_def: Operation name and definition in terms of Commands.
+            **kwargs: Other attributes for the super class.
+        """
+        super().__init__(**kwargs)
+
         self.buffer = buffer
+        self.qubit_freq_est = [freq * 1e9 for freq in qubit_freq_est]
+        self.meas_freq_est = [freq * 1e9 for freq in meas_freq_est]
         self.pulse_library = pulse_library
         self.cmd_def = cmd_def
+        self.ops_def = InstructionScheduleMap(cmd_def, pulse_library)
```

Comment:
This name is up for discussion

Reply:
Some suggestions, `instruction_map`, `instr_map`, `circuit_instr_map` and `circuit_instruction_map` with respective classes `InstructionMap`, `InstrMap`, `CircuitInstrMap` and `CircuitInstructionMap`.